std/stdio.d

```diff
@@ -4522,70 +4521,70 @@ Initialize with a message and an error code.
     }
 }
 
-extern(C) void std_stdio_static_this()
+// Undocummented but public because std* handles are aliasing it
+ref File makeGlobal(alias handle)()
 {
-    static import core.stdc.stdio;
-    //Bind stdin, stdout, stderr
-    __gshared File.Impl stdinImpl;
-    stdinImpl.handle = core.stdc.stdio.stdin;
-    .stdin._p = &stdinImpl;
-    // stdout
-    __gshared File.Impl stdoutImpl;
-    stdoutImpl.handle = core.stdc.stdio.stdout;
-    .stdout._p = &stdoutImpl;
-    // stderr
-    __gshared File.Impl stderrImpl;
-    stderrImpl.handle = core.stdc.stdio.stderr;
-    .stderr._p = &stderrImpl;
+    static __gshared File.Impl impl;
+    static __gshared File result;
+    static shared bool initialized;
+    if (!initialized)
+    {
+        // Use double-checking because initOnce is inefficient.
```

Comment:
@klickverbot I think the optimization is valid, because `initialized` will never go from `true` to `false`. A similar technique could be employed inside `initOnce`, but I think it still doesn't help because of the way initialization is assumed.